File: src/models/resource.ts
import type { ResourceContent } from "./resource-content";
import type { Subject } from "./subject";

export type Resource = Readonly<{
  id: string;
  startDate: Date;
  endDate: Date;
  subject: Subject;

  haveAssignment: boolean;
  assignmentDeadline?: Date;

  /** Color of the resource in HEX. */
  backgroundColor: string;
  contents: Array<ResourceContent>;
  teachers: string[];
  groups: string[];
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/Resource.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class Resource(
    val id: String,
    val startDate: LocalDateTime,
    val endDate: LocalDateTime,
    val subject: Subject,

    val haveAssignment: Boolean,
    val assignmentDeadline: LocalDateTime? = null,

    /** Color of the resource in HEX. */
    val backgroundColor: String,
    val contents: List<ResourceContent>,
    val teachers: List<String>,
    val groups: List<String>
)